packages/gamut/src/Form/FormWrapper.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { FormProvider, Mode, SubmitHandler, useForm } from 'react-hook-form';
+
+import { Form } from '../Form';
+import { FormValues } from './types';
+
+export type FormWrapperProps<Values extends {}> = {
+  children?: React.ReactNode;
+
+  /**
+   * Function called with field values on submit, if all validations have passed.
+   */
+  onSubmit: SubmitHandler<Values>;
+
+  defaultValues?: FormValues;
+
+  /**
+   * Which react hook form mode we are going to use for validation.
+   * If you use the onChange mode the submit button will be disabled until all
+   * required fields are completed.
+   */
+  validation?: Mode;
+};
+
+// this is an in progress API! please reach out to the web-plat team if you're interested in using it.
+
```

Comment:
This way the doc gets nicely attached in IDEs:

```suggestion
/**
 * This is an in progress API! please reach out to the web-plat team if you're interested in using it.
 */
```

If you want to be _really_ aggressive about it you can add `@deprecated` to give it a strikethrough... might be too aggressive though, hehe